### PR TITLE
feat: refresh customer authentication token on expiration

### PR DESCRIPTION
**Pull request by dhargitai**
## Description
When there is a token set in the user's browser but the stored JWT token is expired in it (ie. they're logged out in BigCommerce) then we try to log them in again with their stored customer ID.

## Related Issue
VBC-515

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: packages/theme/pages/Checkout.vue [hunk 1]
```diff
@@ -102,9 +102,7 @@ export default defineComponent({
     });
 
     onMounted(async () => {
-      if (!user.value) {
-        await loadUser();
-      }
+      await loadUser();
```
Comment (scernisevs) [inline packages/theme/pages/Checkout.vue:105]: sorry @davidhargitai-born , just wanted to ask why the condition was removed?

Comment (dhargitai) [reply packages/theme/pages/Checkout.vue:105]: If we try to load the user data every time on loading the checkout page, then we can be sure that they are reauthenticated if their token is already expired at the moment of page load.

File: packages/api-client/src/api/customers/login/index.ts [hunk 2]
```diff
@@ -26,7 +26,8 @@ export const loginCustomer: Endpoints['loginCustomer'] = async (
   params
 ) => {
   try {
-    const loginResponse = await Login.performLogin(context, params);
+    const loginResponse = await Login.getCustomerIdByCredentials(context, params);
```
Comment (WojtekTheWebDev) [inline packages/api-client/src/api/customers/login/index.ts:29]: Can you change the name of const to be more descriptive?  I think it left from previous implementation

**Review by WojtekTheWebDev — CHANGES_REQUESTED** [review]
One small change request from me

**Review by scernisevs — APPROVED** [review]

File: packages/api-client/src/api/customers/login/index.ts (hunk 2 above)
Comment (dhargitai) [reply packages/api-client/src/api/customers/login/index.ts]: Thanks @WojtekTheWebDev, I renamed this to `validationResponse` (along with the method name).

**Review by WojtekTheWebDev — APPROVED** [review]
LGTM

**Pull request merged**